internal/component/loki/source/docker/runner.go

```diff
@@ -95,23 +98,24 @@ func newTailer(l log.Logger, task *tailerTask) *tailer {
 }
 
 func (t *tailer) Run(ctx context.Context) {
-	ch, chErr := t.opts.client.ContainerWait(ctx, t.target.Name(), container.WaitConditionNextExit)
```

Comment:
Previously, the Run will exit when ctx is cancelled or when there's an error from ContainerWait. When exiting, it calls target.Stop. 

Now, we don't seem to call target.Stop, only ticker.Stop. I wonder if this can lead to a resource leak when you update targets. Could you verify this?

Reply:
Thanks for catching this one, I forgot it. I believe that it would leak in some very specific edge cases where the container is still running and producing logs but it got removed from the targets. Calling stop makes sense